app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ShoppingListController < ApplicationController
+  include ShoppingListHelper
+  def index
+    @recipes = Recipe.where(user_id: current_user.id)
+    @foods = Food.where(user_id: current_user.id)
+
+    @required_foods = @recipes.flat_map do |recipe|
+      recipe.recipe_foods.includes(:food).map do |recipe_food|
+        food = recipe_food.food
+        food[:quantity] = recipe_food.quantity
+        food
+      end
+    end
```

Comment:
### [OPTIONAL]
This approach can be considered reasonable depending on the requirements of the application.

The code uses includes(:food) to eager load the associated Food records for better performance. This helps avoid the N+1 query problem by reducing the number of database queries. 👌

Reply:
Thanks for your comments and useful advice!